### PR TITLE
Globalize jquery enhanced

### PR DESCRIPTION
**Pull request by marcoscaceres**
See commit list... I'm going to break this up into several PRs where I can.

### CODE REVIEW

File: js/core/jquery-enhanced.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
+define(["jquery","core/utils"], function($, utils) {
```
Comment (marcoscaceres) [inline js/core/jquery-enhanced.js]: This is a straight copy and paste from core/utils (where this code used to live). This code does not require review.

**Comment by marcoscaceres** [comment]
Actually, I'm just going to annotate the code. As it's not as many changes as it looks.

File: js/core/utils.js [hunk 2]
```diff
@@ -5,175 +5,8 @@
...
-                    for (var i = 0, len = node.childNodes.length; i < len; ++i) getTextNodes(node.childNodes[i]);
-                }
-            }
-            getTextNodes(this[0]);
-            return textNodes;
-        };
-
-
+    [],
+    function () {
```
Comment (marcoscaceres) [inline js/core/utils.js]: This is where I ripped out the additions to JQuery from.

File: tests/SpecRunner.html [hunk 3]
```diff
@@ -10,51 +10,33 @@
...
-    <script src="spec/w3c/rdfa-spec.js"></script>
-    <script src="spec/w3c/style-spec.js"></script>
-
-    <!-- SOURCE -->
-    <!-- none currently -->
+    <script src='/node_modules/requirejs/require.js'></script>
+    <script src="/node_modules/whatwg-fetch/fetch.js"></script>
     <script>
-     jasmine.DEFAULT_TIMEOUT_INTERVAL = 1220000;
+    "use strict";
```
Comment (marcoscaceres) [inline tests/SpecRunner.html]: Here, I use fetch to get the "testFile.json" list to test against.

File: tests/spec/core/jquery-enhanced-spec.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
+"use strict";
```
Comment (marcoscaceres) [inline tests/spec/core/jquery-enhanced-spec.js]: These tests used to live in utils-spec.js, they don't require review.

**Comment by marcoscaceres** [comment]
Ok, finished annotations... ready for review.

**Comment by halindrome** [comment]
I *am* reviewing this. Trying to be careful though.

**Comment by marcoscaceres** [comment]
@halindrome really appreciate your time, thoroughness, help, and support! Your ideas are vastly improving what we are doing here.

**Comment by halindrome** [comment]
Question - do you think this version fixes the general problem that our utils.js methods are not exposed to jquery extended objects when they are used in extensions? 'Cause my testing shows that getDfnTitles, for example, is not a method on a jquery extended object in a JS file loaded by a respec-enabled page.

**Comment by marcoscaceres** [comment]
@halindrome, it should fix it. It literally just puts the extended jQuery into the window object before any of the plugins run.

File: js/core/jquery-enhanced.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+      if (node.nodeType === 1 && excl[node.localName.toLowerCase()]) return;
+      if (node.nodeType === 3) textNodes.push(node);
+      else {
+        for (var i = 0, len = node.childNodes.length; i < len; ++i) getTextNodes(node.childNodes[i]);
+      }
+    }
+    getTextNodes(this[0]);
+    return textNodes;
+  };
+  window.$ = $;
```
Comment (marcoscaceres) [inline js/core/jquery-enhanced.js]: @halindrome here be the magic!!!!! :star2:

**Comment by marcoscaceres** [comment]
@halindrome, see: https://github.com/w3c/respec/pull/623/files#diff-2d019cf669f32febfab70c89006e39caR171

**Comment by marcoscaceres** [comment]
An extension can also `define(["jquery-enhanced"])` ... I even think that is they ask for just "jquery", they will also get the enhanced, as the enhanced one is just enhancing the jQuery singleton.

**Comment by halindrome** [comment]
I think we are talking past one another.  Before we started messing with this, tne utils.js module pushed a bunch of methods into the jquery function prototype so that they were all added to any jquery objects that are later created.  So, when data-oninclude is defined that method will be called and jquery already has the extensions that permit easy integration with respec.  That is not currently happening in this branch.  

Extension scripts don't use require, and in any event there are tons of them out there.  We can't ask them to all go change something in order to start working again.

**Comment by halindrome** [comment]
Just to be clear - these extensions don't ASK for anything.  They are just snippets of JS that get run within the context of ReSpec because it just works.

**Comment by marcoscaceres** [comment]
> Just to be clear - these extensions don't ASK for anything. They are just snippets of JS that get run within the context of ReSpec because it just works.

Understood. They don't need to ask. It will just work* (<small>terms and conditions apply</small>).

* so long as they at least wait for `isRespecReady` or use the `respecEvents` thing.

**Comment by halindrome** [comment]
Well - in the case where I am testing my method is called from the data-oninclude handler and the methods are not htere.  Check out http://dev.spec-ops.io/webpayments-ig/VCTF/use-cases - it loads from this branch's build of respec.

**Comment by marcoscaceres** [comment]
@halindrome, that version of ReSpec seems to be wrong (or didn't build?). It's missing this code. See, in particular, "profile-w3c-common.js". The code from this PR is not there?

**Comment by halindrome** [comment]
yikes..... weird.  testing

**Comment by marcoscaceres** [comment]
@halindrome, in particular: 
```
deps: ["core/jquery-enhanced", "Promise"]
```
is not there. And neither is the `core/jquery-enhanced.js` file.

**Comment by marcoscaceres** [comment]
In other news! YAY! the sourcemap work just came in handy... in even other news, it's 4:30 am, of shit I better got to sleep! :+1:

**Comment by marcoscaceres** [comment]
* oh

**Comment by marcoscaceres** [comment]
Just to show it works for me locally (everything is there):

screenshot 2016-03-10 04 35 36

Did you do? `./tools/build-w3c-common.js` :zzz: :sleeping:

**Comment by halindrome** [comment]
Okay - no idea why it had not built when I changed branches.  I do note that it is complaining about trying to load node_modules/promise-polyfill/Promise.js from the built version.  I will try adding that to collection of required components that need to be included in the build.  But yes, I can confirm that the object is extended!

Get some sleep.

**Comment by marcoscaceres** [comment]
>  I do note that it is complaining about trying to load node_modules/promise-polyfill/Promise.js from the built version.

Can you please post a new bug with the error? That should only happen if it's missing from your `node_modules` dir, or something is not right in the builder. I'm not seeing that error locally here.

**Comment by halindrome** [comment]
I will do.  I do see that there is a dependency on it in that core/jquer-enhanced requires Promise, but we never load hqyer-enhanced in the profile-w3c-common.js file

**Comment by halindrome** [comment]
Oh - and otherwise LGTM.

**Comment by marcoscaceres** [comment]
Confirmed... it is going to the network for Promise.js. My thought is to just remove it, as all the browsers we target support it: http://caniuse.com/#search=Promise

**Comment by halindrome** [comment]
Huh.    Remember people use this for editors drafts and we have no control at all who revirews those.  Including people who use older  user agents.  ReSpec - it's not just for development any more. (tm)

**Comment by marcoscaceres** [comment]
@halindrome, sure, but we have to cut off somewhere. Promises have been natively supported in browsers for over a year, and all browsers are now evergreen. It seems highly unlikely people reading specs (specially ed drafts) won't be running a browser that is at most 3 months old.

**Comment by halindrome** [comment]
I am willing to try this - I just note for the record that it could bite us in the ass.  In particular with Windows users who have not moved to Windows 10 and for whatever reason use IE.

**Comment by marcoscaceres** [comment]
Let's see what happens. If it's bad, we can quickly add the poly back.

**Comment by halindrome** [comment]
Assuming this passes I am comfortable with you merging and pushing a new release.

**Comment by marcoscaceres** [comment]
Ha! I think ./tests/headless.js just ruined the party.

**Comment by halindrome** [comment]
you're right.  I had this problem the other day too.  UGGH

**Comment by marcoscaceres** [comment]
Backed out the last commit.

**Comment by marcoscaceres** [comment]
Ok, will merge this after it turns green and figure out why Promise is not being inserted tomorrow (unless you want to have a look). The `respecIsReady` promise is also ready to merge, I hope.

**Comment by halindrome** [comment]
works for me.  I will play with it a little in a new (local) branch once it is all merged.

**Pull request merged**